internal/worker/instancemutater/mutater.go

```diff
@@ -230,7 +230,8 @@ func (m MutaterMachine) processMachineProfileChanges(info *instancemutater.UnitP
 		return report(errors.Annotatef(err, "%s", m.id))
 	}
 
-	expectedProfiles := m.context.getRequiredLXDProfiles(info.ModelName)
+	modelProfileName := fmt.Sprintf("%s-%s", info.ModelName, names.NewModelTag(info.ModelUUID).ShortId())
```

Comment:
So here, info may or may not contain the model uuid, depending on what version of the facade we were talking to. The caller needs to make a decision as to whether the old or new naming is used depending on the facade version.